pkg/stanza/fileconsumer/internal/tracker/tracker.go

```diff
@@ -106,6 +131,7 @@ func (t *fileTracker) ClosePreviousFiles() {
 
 	for r, _ := t.previousPollFiles.Pop(); r != nil; r, _ = t.previousPollFiles.Pop() {
 		t.knownFiles[0].Add(r.Close())
+		t.openFiles.Add(context.TODO(), -1)
```

Comment:
This is really the only opening or closing of files we do in the tracker, so it does feel quite awkward to have the tracker manage the meter. Instead, what if we just return the number of files closed from this method? We're closing them one after another, so decrementing after each one or all at once seems equivalent in this case.

Reply:
In that case we would need to either have the [`EndConsume()`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.100.0/pkg/stanza/fileconsumer/internal/tracker/tracker_other.go#L15) to return the closed files as well or to move the `ClosePreviousFiles()` outside of it. I'm leaning towards the latter.

Reply:
The latter won't work because the order of operations is different depending on OS.

I think we could get rid of OS dependent behavior for _closing_ files if we implement https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32037.

Otherwise, I'm fine with returning the count from `EndConsume`.